x/rewards/types/msg.go

```diff
@@ -212,3 +214,44 @@ func (m MsgSetFlatFee) ValidateBasic() error {
 
 	return nil
 }
+
+// NewMsgUpdateParams creates a new MsgUpdateParams instance.
+func NewMsgUpdateParams(senderAddr sdk.AccAddress, params Params) *MsgUpdateParams {
+	msg := &MsgUpdateParams{
+		Authority: senderAddr.String(),
+		Params:    params,
+	}
+
+	return msg
+}
+
+// Route implements the sdk.Msg interface.
+func (m MsgUpdateParams) Route() string { return RouterKey }
+
+// Type implements the sdk.Msg interface.
+func (m MsgUpdateParams) Type() string { return TypeMsgFlatFee }
+
+// GetSigners implements the sdk.Msg interface.
+func (m MsgUpdateParams) GetSigners() []sdk.AccAddress {
+	senderAddr, err := sdk.AccAddressFromBech32(m.Authority)
+	if err != nil {
+		panic(fmt.Errorf("parsing sender address (%s): %w", m.Authority, err))
+	}
+
+	return []sdk.AccAddress{senderAddr}
+}
+
+// GetSignBytes implements the sdk.Msg interface.
+func (m MsgUpdateParams) GetSignBytes() []byte {
```

Comment:
this is for amino so we don't need it, since this will not be run through a TX, especially amino tx